frontend/simple/js/events.js

```diff
@@ -44,6 +65,17 @@ export class GroupContract extends Events.HashableGroup {
           await backend.subscribe(identityContractId)
           await store.dispatch('syncContractWithServer', identityContractId)
         }
+      },
+      async  HashableGroupVoteForOrAgainstProposal ({state}, data) {
+        if (store.state.loggedIn !== data.username && !state.proposals[data.proposalHash]) {
```

Comment:
This condition here makes it possible that the code below won't get executed.

There's also another condition on L72 which means there's another place where "the right thing" might not happen.

- This is problematic because it means if there's any error that happens then we are not able to ever recover again, since the async actions are immediately cleared after the loop.
- I also don't see where any action is taken on the proposal itself besides deleting this message...